internal/infrastructure/kubernetes/infra_client.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package kubernetes
+
+import (
+	"context"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type InfraClient struct {
```

Comment:
I did not like this personally, this's not clear to me then before.

Reply:
I'd like something like https://github.com/GoogleContainerTools/kpt-config-sync/tree/main/pkg/applier

Reply:
current `InfraClient` is different, it's just a  subclass or extension for `client.Client` to implement `create` and `delete` resource logic.

```golang
// Copyright Envoy Gateway Authors
// SPDX-License-Identifier: Apache-2.0
// The full text of the Apache license is available in the LICENSE file at
// the root of the repo.

package applier

import (
	"context"
	"fmt"
	"reflect"

	appsv1 "k8s.io/api/apps/v1"
	corev1 "k8s.io/api/core/v1"
	kerrors "k8s.io/apimachinery/pkg/api/errors"
	"k8s.io/apimachinery/pkg/types"
	"sigs.k8s.io/controller-runtime/pkg/client"
)

type Instance struct {
	Client client.Client
}
```

```golang
type InfraClient struct {
	client.Client
}
```

Reply:
so that, rename of the `pkg` or `struct` seems meaningless to me personally.

Reply:
it not only renames but also refactors the `CreateXX` and `DeleteXX` logic to single `Create` and `Delete` `Service`,`ConfigMap`, `Deployment`, `ServiceAccount`